staging/src/k8s.io/kubectl/pkg/cmd/config/set.go

```diff
@@ -141,6 +202,311 @@ func (o setOptions) validate() error {
 	return nil
 }
 
+// modifyConfigJson: modifyConfig, but for JSON path
+func modifyConfigJson(config *clientcmdapiv1.Config, propertyName, propertyValue string, unset bool, setRawBytes bool, deduplicate bool) error {
+	// Create jsonpath parser to use throughout the function
+	jsonPath := jsonpath.New("Value Getter").AllowMissingKeys(true)
+
+	jsonPathParser, err := jsonpath.Parse("Node Parser", propertyName)
```

Comment:
Why not call `FindResults` on this instance of `jsonPathParser` instead of the complicated logic below?